---
uint64 size
bool is_empty
